editor/create_dialog.cpp

```diff
@@ -246,9 +246,9 @@ void CreateDialog::add_type(const String &p_type, HashMap<String, TreeItem *> &p
 		to_select_type = to_select_type.split(" ")[0];
 		bool current_item_is_preffered;
 		if (cpp_type) {
-			current_item_is_preffered = ClassDB::is_parent_class(p_type, preferred_search_result_type) && !ClassDB::is_parent_class(to_select_type, preferred_search_result_type);
+			current_item_is_preffered = ClassDB::is_parent_class(p_type, preferred_search_result_type) && !ClassDB::is_parent_class(to_select_type, preferred_search_result_type) && search_box->get_text() != to_select_type;
```

Comment:
Could you also fix the typo with `current_item_is_preffered`? (should be `preferred`)

Reply:
Fixed